File: cs/2348_Number_of_Zero-Filled_Subarrays.cs
public class Solution {
    public long ZeroFilledSubarray(int[] nums) {
        long totalZeroSubarrays = 0;
        long currentZeroSubarrays = 0;
                
        for (int i = 0; i < nums.Length; ++i) {
            if (nums[i] == 0) {
                currentZeroSubarrays++;
                totalZeroSubarrays += currentZeroSubarrays;
            } else {
                currentZeroSubarrays = 0;
            }
        }  
           
        return totalZeroSubarrays;
    }
}

File: php/2348_Number_of_Zero-Filled_Subarrays.php
<?php

class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function zeroFilledSubarray($nums)
    {
        $totalZeroSubarrays = $currentZeroSubarrays = 0;

        foreach ($nums as $num) {
            if ($num == 0) {
                $currentZeroSubarrays++;
                $totalZeroSubarrays += $currentZeroSubarrays;
            } else {
                $currentZeroSubarrays = 0;
            }
        }

        return $totalZeroSubarrays;
    }

}
